x-pack/platform/plugins/shared/ml/public/alerting/anomaly_detection_rule/ml_anomaly_alert_trigger.tsx

```diff
@@ -137,6 +147,30 @@ const MlAnomalyAlertTrigger: FC<MlAnomalyAlertTriggerProps> = ({
     [jobsAndGroupIds]
   );
 
+  useEffect(
+    function handleFilterPreservationAcrossResultTypes() {
+      const isBucketType = ruleParams.resultType === ML_ANOMALY_RESULT_TYPE.BUCKET;
+
+      if (isBucketType) {
+        if (ruleParams.customFilter) {
+          setSavedFilterForNonBucketTypes(ruleParams.customFilter);
+          setRuleParams('customFilter', null);
+
+          setKqlFieldUsage({
+            hasAnomalyScoreFilter: false,
+            hasInterimFilter: false,
+          });
+        }
+      } else {
+        if (savedFilterForNonBucketTypes && !ruleParams.customFilter) {
+          setRuleParams('customFilter', savedFilterForNonBucketTypes);
```

Comment:
I'm finding that alerts aren't triggering if I don't have a filter set (Kibana sample data logs, logs_response_code_rates job, record type). Have you seen alerts with no filter?